dockerfiles/cpp-23.Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM gcc:13.2.0-bookworm
+
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="vcpkg.json,vcpkg-configuration.json"
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y zip=3.* && \ 
+    apt-get install --no-install-recommends -y g++=4:* && \
+    apt-get install --no-install-recommends -y build-essential=12.* && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# cmake 3.29.2 is required by vcpkg
+RUN wget --progress=dot:giga https://github.com/Kitware/CMake/releases/download/v3.29.2/cmake-3.29.2-Linux-x86_64.tar.gz && \
+    tar -xzvf cmake-3.29.2-Linux-x86_64.tar.gz && \
+    mv cmake-3.29.2-linux-x86_64/ /cmake
+
+ENV CMAKE_BIN="/cmake/bin"
+ENV PATH="${CMAKE_BIN}:$PATH"
+
+RUN git clone https://github.com/microsoft/vcpkg.git && \
+    ./vcpkg/bootstrap-vcpkg.sh
+
+ENV VCPKG_ROOT="/vcpkg"
+ENV PATH="${VCPKG_ROOT}:$PATH"
+
+WORKDIR /app
+
+COPY vcpkg.json ./
+COPY vcpkg-configuration.json ./
```

Comment:
We are now copying the full directory in a couple of steps, instead of just the dependency files. 
Ref: https://github.com/codecrafters-io/build-your-own-git/pull/96

Reply:
Got it!